### PR TITLE
refactor: Update CI script for pyfluent release.

### PR DESCRIPTION
**Pull request by prmukherj**
version 24.1 has been moved as part of the Unit Testing in CI along with other earlier versions. 
only 24.2 is kept as part of separate build and testing.
Build matrix has been updated to include 24.1 as well.

### CODE REVIEW

**Comment by prmukherj** [comment]
@mkundu1, could you please have a look at the current workflow and provide your comments if something needs modification.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -350,6 +350,40 @@ jobs:
...
+      - name: Run 24.1 API codegen
+        if: steps.cache-241-api-code.outputs.cache-hit != 'true'
+        run: make api-codegen
+        env:
+          FLUENT_IMAGE_TAG: v24.1.0
+
+      - name: Print 24.1 Fluent version info
+        run: |
+          cat src/ansys/fluent/core/fluent_version_241.py
+          python -c "from ansys.fluent.core.solver.settings_232 import SHASH; print(f'SETTINGS_HASH = {SHASH}')"
```
Comment (mkundu1) [inline .github/workflows/ci.yml:385]:
```suggestion
          python -c "from ansys.fluent.core.solver.settings_241 import SHASH; print(f'SETTINGS_HASH = {SHASH}')"
```

Comment (prmukherj) [reply .github/workflows/ci.yml:385]: Thank you @mkundu1

**Comment by mkundu1** [comment]
> @mkundu1, could you please have a look at the current workflow and provide your comments if something needs modification.

The workflow changes look good to me, thanks.

**Review by mkundu1 — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

**Pull request merged**